### PR TITLE
[CONTP-962] Add tests for IAM role names with paths.

### PR DESCRIPTION
**Pull request by Mathew-Estafanous**
### What does this PR do?

PR https://github.com/DataDog/terraform-aws-ecs-datadog/pull/40 fixed role arn name parsing to account for a role's `path`. This PR adds unit tests to validate that change/

### Motivation

Improve test coverage for edge cases like the one identified in the merged PR.

### Describe how you validated your changes

- [x] CI is green 🟢

### CODE REVIEW

File: tests/role_parsing_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+	err := json.Unmarshal([]byte(task["container_definitions"]), &containers)
+	s.NoError(err, "Failed to parse container definitions")
+
+	s.NotEmpty(task["arn"], "Task definition ARN should not be empty")
+	s.NotEmpty(task["revision"], "Task definition revision should not be empty")
+
+	taskRoleArn := task["task_role_arn"]
+	s.NotEmpty(taskRoleArn, "Task role ARN should not be empty")
+	s.Contains(taskRoleArn, "/terraform-test/", "Task role ARN should contain the path '/test-path/'")
```
Comment (gabedos) [inline tests/role_parsing_test.go:34]: nit: expected role path
```suggestion
	s.Contains(taskRoleArn, "/terraform-test/", "Task role ARN should contain the path '/terraform-test/'")
```

**Review by gabedos — APPROVED** [review]
Approved with small comment. Thanks for adding these tests!
